api/src/main/java/io/strimzi/api/kafka/model/kafka/cruisecontrol/CruiseControlSpec.java

```diff
@@ -60,7 +60,7 @@ public class CruiseControlSpec implements HasConfigurableMetrics, HasConfigurabl
     private BrokerCapacity brokerCapacity;
     private Map<String, Object> config = new HashMap<>(0);
     private MetricsConfig metricsConfig;
-    private Map<String, Object> additionalProperties = new HashMap<>(0);
+    private final Map<String, Object> additionalProperties = new HashMap<>(0);
```

Comment:
I think in general, we do not set these as `final` (although some of the additionalProperties are final, it seems to be the minority). I wonder if we should aim for some unification (one way or another - I do not have any strong opinion on what is correct).

Reply:
I think it should be final everywhere, but I opened this PR just for Cruise Control related stuff I noticed while working on auto-rebalancing. This PR doesn't aim to fix all the `final` missing around the codebase.

Reply:
any more on this @scholzj ?

Reply:
For me the PR is ok as it is in its narrow scope. I went just through the Cruise Control stuff and this field makes sense to be `final`.

Reply:
Well, feel free to merge it if you want. But you seem to be diverging from the _common way_ it is used in our code. So I think it would be great if you had a plan to fix it everywhere and not just in the two files (not necessarily in this PR, but at least later, or open an issue for it or something).

Reply:
AFAICS, `KafkaNodePoolTemplate`, `KafkaBridge`, `PasswordSource`, `KafkaConnect`, `AutoRestart`, `KafkaConnector`, `Kafka` and more are some classes which are defining this map as `final`. All others are not. So I don't think this PR should address this comment. I can  open a different PR/issue for that.

Reply:
I can follow up with the task of making additionalProperties final in all spec classes.

Reply:
Thanks Fede, going to merge this one then.